### ISSUE TITLE
Language features not working with multi projects

### ISSUE THREAD

**Issue by manlao** [graphql/graphiql#2347]

**Comment by HosseinAgha** [comment]
I'm having the same issue using the `graphql.config.yml` instead of legacy `.graohqlrc.yml`.  
Looks like only the first schema will be used by GraphQL Language Server.

**Comment by avele** [comment]
It still functions correctly only for the first project in the list, but incorrectly validates the remaining projects using the schema of the first project

**Comment by acao** [comment]
@avele correct, i found a fix to this and several other issues this weekend

**Comment by HosseinAgha** [comment]
@acao could you manage to commit the fix?  
Can we contribute? Could you please provide the solution in your mind?

**Comment by EduardoSaavedra** [comment]
any news??

**Comment by acao** [comment]
hey sorry, I will pivot back to this soon. The branch had a ton of other bugs and I then went on to try to build an e2e suite for the extension to detect whether they were regressions and hit a bunch of roadblocks. I think a suite of solid high level integration tests should allow me to establish the known or unknown bugs, and then I can guarantee we can fix this in a way that works without introducing regressions across a wide range of project configs, across schema configs and pertinent configurations like persisting schema from network, etc. 

The only issue with this over proper vscode extension e2e suite is that by mocking the IDE extension client messages, we don't always get the full spectrum of realistic client behavior. I would like to solve this before introducing multi-root workspace support, because that compounds the multi-project complexity, though I vaugely recall the contributed PR for multi-root effort may have solved this issue as well.

I should also probably separate out the part where I re-validate the entire project file tree on change (thus the issues dissapear and the red files, as with the typescript service in the last few years), or make it a config option, but the more ideal solution would be an approach that only re-validates files that contain related graphql operations or schema. I manage to debounce on change, but validating the entire document tree on change is a lot of work!

**Comment by martinezcoder** [comment]
I have the same problem in a completely different environment, I suppose the library is shared.

## Specifications

NVIM v0.9.5
Using Mason graphql-language-service-cli version 3.5.0
